lib/excerpt_parser.rb

```diff
@@ -114,7 +114,11 @@ def start_element(name, attributes = [])
       @in_quote = !@keep_onebox_source if attributes.include?(%w[class source])
     when "div", "span"
       attributes = Hash[*attributes.flatten]
-      if attributes["class"]&.include?("excerpt")
+
+      # Only match "excerpt" class if it does specifically equal "excerpt
+      # hidden" in order to prevent internal links with GitHub oneboxes from
+      # being empty https://meta.discourse.org/t/269436
+      if attributes["class"]&.include?("excerpt") && !attributes["class"]&.match?("excerpt hidden")
```

Comment:
For GitHub oneboxes I think it will always be in this order.

<img width="749" alt="image" src="https://github.com/discourse/discourse/assets/1490496/d40c46bd-5a60-4eb6-9cea-7158267bc1dc">

I suppose some other onebox could have a different class order though.

Reply:
I just tried Wikipedia and some other oneboxes and they don't have expand more links. I'm fine merging this in as is, and updating it case-by-case bases for any specific oneboxes that are adding the `excerpt` class which I think is conflicting with some core `excerpt` class